velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -340,7 +350,13 @@ bool registerAverageAggregate(const std::string& name) {
           checkSumCountRowType(
               inputType,
               "Input type for final aggregation must be (sum:double, count:bigint) struct");
-          return std::make_unique<AverageAggregate<int64_t>>(resultType);
+          if (resultType->kind() == TypeKind::DOUBLE) {
+            return std::make_unique<AverageAggregate<int64_t, double, double>>(
+                resultType);
+          } else {
```

Comment:
Should we use a switch(resultType->kind()) with DOUBLE and REAL cases implemented and default using VELOX_FAIL?

Reply:
Would it be more robust to explicitly handle resultType == REAL and fail if resultType is neither DOUBLE nor REAL?

Reply:
It makes sense that a default fail is more robust.